### ISSUE TITLE
YFA Workflow doesn't prompt for Yubikey touch

### ISSUE THREAD

**Issue by Gowiem** [robertoriv/alfred-yubikey-otp#9]
Follow up with a new issue from #6. 

Using `v1.1.1`

Flow: 

1. Type `yfa ` into alfred
2. Search one of my oath names by typing in `oc`. Two search results are shown: `oc` + `oc-tf`
3. Hit enter
4. Alfred prompt very briefly (less than 1 second) shows "Touch your yubikey..."
5. Alfred then reverts back to the search results view of `oc` + `oc-tf`. 
6. If I hit enter at this point, then `oc` is copied into my clipboard. Same if I hit enter on `oc-tf`. 

Odd thing is that this works for other oath entries that are not `oc`... 

Can share logs directly if that is helpful.

**Comment by robertoriv** [comment]
Hi @Gowiem,

There are a couple of things I think I'll need to look at. 

First, it could be a timeout issue where your yubi is returning multiple lines saying touch your yubikey. We handle that with a simple counter: https://github.com/robertoriv/alfred-yubikey-otp/blob/master/src/yubi.py#L123. If we got more than one entry from the search command (`ykman oath accounts code <search term>`), that would count up and display a timeout error (which you haven't specifically mentioned).

Second, I've seen some weird behavior with similarly-named entries (for example: `msg1`, `msg2`, `msg3`) where the fuzzy search in the cache returns more than one entry.

It'll be a few days before I can sit down and play around with this but, if you have logs you are willing to share, that would be helpful.

**Comment by Gowiem** [comment]
@robertoriv got it -- Thanks for the info. I'd guess it was the latter of the possibilities that you mentioned as I have multiple entries and only the ones with similar names are causing me issues I believe. I'm not sure what is in the logs, so I'd be happy to share them with you, but would you mind if I send them over email? Let me know -- Thanks!

**Comment by robertoriv** [comment]
Hi @Gowiem,

Ended up reproducing the issue by creating some fake `oc` and `oc-tf` entries.

Turns out it was due, in part, by Alfred-Workflow's filter implementation (and my misunderstanding of it). It was returning multiple partial matches, instead of a single one.

I've added an (ugly) implementation of an `exact_match` search. I'll need to clean things up later, but it seems to fix the `oc`/`oc-tf` issue.

Give [v1.2.0](https://github.com/robertoriv/alfred-yubikey-otp/releases/tag/v1.2.0) a shot, and let me know if it fixes things for you.
